chsdi/views/catalog.py

```diff
@@ -2,13 +2,52 @@
 
 from pyramid.view import view_config
 from pyramid.httpexceptions import HTTPNotFound
+import networkx as nx
+from itertools import chain
 
 from chsdi.models.bod import Catalog
 from chsdi.lib.validation import MapNameValidation
 from chsdi.lib.sqlalchemy_customs import remove_accents
 from chsdi.lib.filters import filter_by_geodata_staging
 
 
+def tree_data(G, root, attrs, meta):
+    if G.number_of_nodes() != G.number_of_edges() + 1:
+        raise TypeError("G is not a tree.")
+    if not G.is_directed():
+        raise TypeError("G is not directed.")
+
+    id_ = attrs['id']
+    children = attrs['children']
+    if id_ == children:
+        raise nx.NetworkXError('Attribute names are not unique.')
+
+    def add_children(n, G):
+        nbrs = G[n]
+        if len(nbrs) == 0:
+            return []
+        children_ = []
+        for child in nbrs:
+            d = dict(chain(G.node[child].items(), [(id_, child)]))
+            for key in meta:
+                d['category'] = meta[d['id']]['category']
+                d['staging'] = meta[d['id']]['staging']
+                d['label'] = meta[d['id']]['label']
+                if (d['category'] == 'layer'):
+                    d['layerBodId'] = meta[d['id']]['layerBodId']
+                else:
+                    d['selectedOpen'] = meta[d['id']]['selectedOpen']
+            c = add_children(child, G)
+            if c:
+                d[children] = c
+            children_.append(d)
```

Comment:
here you could either sort the children according to the order key if not None or use the `insert` method of python lists. To insert at specific location.